import "dos.proto";
package dos;

message WriteLayerRequest {
  optional string id = 1;
  optional uint32 seq_id = 2;
  optional bytes data = 3;
  optional uint64 offset = 4;
  optional uint64 size = 5;
}

message WriteLayerResponse {
  optional RpcStatus status = 1;
  // the size that has been written
  optional uint64 size = 2;
}

service DlfsLet {
  rpc WriteLayer(WriteLayerRequest) returns (WriteLayerResponse);
}
